std/string.d

```diff
@@ -2468,7 +2468,7 @@ if ((hasSlicing!Range && hasLength!Range) ||
     static struct Result
     {
     private:
-        Range _input;
+        StringTypeOf!Range _input;
```

Comment:
I tested with byUTF!dchar and lineSplitter seamed to be broken even without my patch.

typeof("str".byUTF!dchar) does not fulfill (hasSlicing!Range && hasLength!Range) as the result of byUTF does not have hasSlicing